### PR TITLE
fix #1476 fix serialization in transport layer

### LINKED ISSUE #1476: serialization works incorrectly in cluster mode

### PR DESCRIPTION
**Pull request by huafengw**

### CODE REVIEW

**Comment by stanleyxu2005** [comment]
+1

File: core/src/main/scala/io/gearpump/transport/netty/Client.scala [hunk 1]
```diff
@@ -200,11 +200,11 @@ object Client {
     }
   }
 
-  class ClientPipelineFactory(name: String, decoder: MessageDecoder, encoder: MessageEncoder) extends ChannelPipelineFactory {
+  class ClientPipelineFactory(name: String, conf: NettyConfig) extends ChannelPipelineFactory {
```
Comment (manuzhang) [inline core/src/main/scala/io/gearpump/transport/netty/Client.scala]: I think the original interface is better for testing where decoder and encoder could be mocked. Same holds for ServerPipelineFactory

Comment (stanleyxu2005) [inline core/src/main/scala/io/gearpump/transport/netty/Client.scala]: He wants factory produce encoder decoder internally on demand. There is a thread unsafe issue.

Comment (manuzhang) [inline core/src/main/scala/io/gearpump/transport/netty/Client.scala]: With `conf.newTransportSerializer`, each client has its own serializer for ChannelPipeline. Also, I wonder why a client requires a decoder and server an encoder.

Comment (huafengw) [inline core/src/main/scala/io/gearpump/transport/netty/Client.scala]: There is no test cover for the pipeline factory, but there is a Netty.spec which will test the Client and Server, so the pipeline factory is embedded.

**Pull request merged**